org.wso2.carbon.consent.mgt.core/src/main/java/org/wso2/carbon/consent/mgt/core/dao/JdbcTemplate.java

```diff
@@ -0,0 +1,219 @@
+                queryFilter.filter(preparedStatement);
+            }
+            ResultSet resultSet = preparedStatement.executeQuery();
+            int i = 0;
+            while (resultSet.next()) {
+                T row = rowMapper.mapRow(resultSet, i);
+                result.add(row);
+                i++;
+            }
+        } catch (SQLException e) {
+            logDebugInfo(
+                    "There has been an error performing the database query. The query is {}, and the Parameters are {}",
+                    e, query, queryFilter);
+            throw new DataAccessException(ConfigurationConstants.ErrorMessages.ERROR_CODE_DATABASE_QUERY_PERFORMING
+                    .getMessage() + query, ConfigurationConstants.ErrorMessages.ERROR_CODE_DATABASE_QUERY_PERFORMING
+                    .getCode(),
+                    e);
+        }
+        return result;
+    }
+
+    /**
+     * Executes a query on JDBC and return the result as a domain object.
+     *
+     * @param query       the SQL query with the parameter placeholders.
+     * @param rowMapper   Row mapper functional interface
+     * @param queryFilter parameters for the SQL query parameter replacement.
+     * @return domain object of required type.
+     */
+    public <T extends Object> T fetchSingleRecord(String query, RowMapper<T> rowMapper, QueryFilter queryFilter)
+            throws DataAccessException {
+        T result = null;
+        try (Connection connection = dataSource.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            if (queryFilter != null) {
+                queryFilter.filter(preparedStatement);
+            }
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                result = rowMapper.mapRow(resultSet, 0);
+            }
+            if (resultSet.next()) {
+                logDebugInfo("There are more records than one found for query: {} for the parameters {}", query,
+                        queryFilter);
+                throw new DataAccessException(ConfigurationConstants.ErrorMessages.ERROR_CODE_MORE_RECORDS_IN_QUERY
+                        .getMessage() + query, ConfigurationConstants.ErrorMessages.ERROR_CODE_MORE_RECORDS_IN_QUERY
+                        .getCode());
+            }
+        } catch (SQLException e) {
+            logDebugInfo(
+                    "There has been an error performing the database query. The query is {}, and the parameters are {}",
+                    e, query, rowMapper, queryFilter);
+            throw new DataAccessException(ConfigurationConstants.ErrorMessages.ERROR_CODE_DATABASE_QUERY_PERFORMING +
+                    query, ConfigurationConstants.ErrorMessages.ERROR_CODE_DATABASE_QUERY_PERFORMING.getCode(), e);
+        }
+        return result;
+    }
+
+    public void executeUpdate(String query, QueryFilter queryFilter) throws DataAccessException {
```

Comment:
Shall we add a method comment since its a public method and also add a newline after the method signature?